### PR TITLE
Fix: Proper plugin logo loading depending on staging

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**

Change from hardcoded - always going to grafana.com - to now being relative to whatever is configured in grafana config.

**Why do we need this feature?**
For our staging environment we use different GCOM URLs and this makes logos follow the same convention.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by xnyo** [comment]
I tested this locally and it looks like the `Cache` and `Expires` headers set by GCOM are being stripped/changed.

For example:
http://127.0.0.1:3000/api/gnet/plugins/alertlist/versions/5.0.0/logos/small
Pasted image 20240523191924

The `Expires` header is missing, and `Cache-Control` is replaced with `no-store`.

GCOM does set `Expires` and `Cache-Control`: 
https://github.com/grafana/grafana-com/blob/3b6168cbc192f15820583d065c340a06fc87d032/packages/grafana-com-base/api/plugins/plugin-versions.routes.ts#L427-L429

In fact, if I get the logo directly from GCOM, I can see them:
https://grafana.com/api/plugins/alertlist/versions/5.0.0/logos/small
Pasted image 20240523192140

It looks like they are being stripped here:
https://github.com/grafana/grafana/blob/44e1bce55a27320462c646a1aff577b4abae43a2/pkg/middleware/middleware.go#L50-L56

By this function:
https://github.com/grafana/grafana/blob/44e1bce55a27320462c646a1aff577b4abae43a2/pkg/middleware/middleware.go#L91-L95

I have pushed this commit on a separate branch which should fix it:
https://github.com/grafana/grafana/commit/2441ab4bde58e8475dc521ed26494181f2ded55a


Regardless of the cache, I also ran into a different issue where the GCOM API started returning 429s after I opened the plugin catalog list and clicked "All". This was more evident when the caching headers were removed by the Grafana backend (in that case, refreshing twice would break all icons), but it's also present if the requests don't go through the backend proxy (refreshing twice with CTRL+F5 to ignore cache while on the "All" plugins catalog page breaks all icons)

immagine

Pasted image 20240523193435

The 429 is returned directly by GCOM, not by the proxy (see here: https://ops.grafana-ops.net/goto/Pfv3gEPIg?orgId=1)

It's defined here
https://github.com/grafana/grafana-com/blob/3b6168cbc192f15820583d065c340a06fc87d032/packages/grafana-com-base/api/plugins/plugin-versions.routes.ts#L401-L408, it should probably be bumped :slightly_smiling_face:

**Comment by tolzhabayev** [comment]
Very good points I will play around with different options - see whats best. we want to keep to caching headers ideally intact to prevent 429s.

File: public/app/features/plugins/admin/helpers.ts [hunk 1]
```diff
@@ -126,8 +126,8 @@ export function mapRemoteToCatalog(plugin: RemotePlugin, error?: PluginError): C
     id,
     info: {
       logos: {
-        small: `https://grafana.com/api/plugins/${id}/versions/${version}/logos/small`,
-        large: `https://grafana.com/api/plugins/${id}/versions/${version}/logos/large`,
+        small: `${config.appSubUrl}/api/gnet/plugins/${id}/versions/${version}/logos/small`,
```
Comment (jackw) [inline public/app/features/plugins/admin/helpers.ts:129]: Is this going via the grafana server gnet proxy? Is so do we need `config.appSubUrl`?

Comment (tolzhabayev) [reply public/app/features/plugins/admin/helpers.ts:129]: it does, and yes we do if we run from a subfolder

**Comment by tolzhabayev** [comment]
@xnyo @jackw ok it works, also on the ephemeral instance. build is green, should we merge?

**Review by xnyo — APPROVED** [review]
LGTM! :raised_hands:

**Pull request merged**